sympy/solvers/ode/tests/test_ode.py

```diff
@@ -1446,124 +1446,6 @@ def test_1st_exact2_broken_check():
     assert checkodesol(eq, sol, order=1, solve_for_func=False)[0]
 
 
-def test_separable1():
-    # test_separable1-5 are from Ordinary Differential Equations, Tenenbaum and
-    # Pollard, pg. 55
-    eq1 = f(x).diff(x) - f(x)
-    eq2 = x*f(x).diff(x) - f(x)
-    eq3 = f(x).diff(x) + sin(x)
-    eq4 = f(x)**2 + 1 - (x**2 + 1)*f(x).diff(x)
-    eq5 = f(x).diff(x)/tan(x) - f(x) - 2
-    eq6 = f(x).diff(x) * (1 - sin(f(x))) - 1
-    sol1 = Eq(f(x), C1*exp(x))
-    sol2 = Eq(f(x), C1*x)
-    sol3 = Eq(f(x), C1 + cos(x))
-    sol4 = Eq(f(x), tan(C1 + atan(x)))
-    sol5 = Eq(f(x), C1/cos(x) - 2)
-    sol6 = Eq(-x + f(x) + cos(f(x)), C1)
-    assert dsolve(eq1, hint='separable') == sol1
-    assert dsolve(eq2, hint='separable') == sol2
-    assert dsolve(eq3, hint='separable') == sol3
-    assert dsolve(eq4, hint='separable') == sol4
-    assert dsolve(eq5, hint='separable') == sol5
-    assert dsolve(eq6, hint='separable') == sol6
-    assert checkodesol(eq1, sol1, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq2, sol2, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq3, sol3, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq4, sol4, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq5, sol5, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq6, sol6, order=1, solve_for_func=False)[0]
-
-
-@slow
-def test_separable2():
-    a = Symbol('a')
-    eq6 = f(x)*x**2*f(x).diff(x) - f(x)**3 - 2*x**2*f(x).diff(x)
-    eq7 = f(x)**2 - 1 - (2*f(x) + x*f(x))*f(x).diff(x)
-    eq8 = x*log(x)*f(x).diff(x) + sqrt(1 + f(x)**2)
-    eq9 = exp(x + 1)*tan(f(x)) + cos(f(x))*f(x).diff(x)
-    eq10 = (x*cos(f(x)) + x**2*sin(f(x))*f(x).diff(x) -
-            a**2*sin(f(x))*f(x).diff(x))
-    sol6 = Eq(Integral((u - 2)/u**3, (u, f(x))),
-        C1 + Integral(x**(-2), x))
-    sol7 = Eq(-log(-1 + f(x)**2)/2, C1 - log(2 + x))
-    sol8 = Eq(asinh(f(x)), C1 - log(log(x)))
-    # integrate cannot handle the integral on the lhs (cos/tan)
-    sol9 = Eq(Integral(cos(u)/tan(u), (u, f(x))),
-        C1 + Integral(-exp(1)*exp(x), x))
-    sol10 = Eq(-log(cos(f(x))), C1 - log(- a**2 + x**2)/2)
-    assert dsolve(eq6, hint='separable_Integral').dummy_eq(sol6)
```

Comment:
@oscarbenjamin this example is tested with _integral hint but if I test it with only separable hint the output is like this :
```

AssertionError: Different solution found from dsolve for example separable_07.

The ODE is:
x**2*f(x)*Derivative(f(x), x) - 2*x**2*Derivative(f(x), x) - f(x)**3

The expected solution was
[Eq(f(x), (-x + sqrt(x*(4*C1*x + x - 4)))/(2*C1*x - 2)), Eq(f(x), (-x - sqrt(x*(4*C1*x + x - 4)))/(2*C1*x - 2))]

What dsolve returned is:
[Eq(f(x), (-x + sqrt(x*(4*C1*x + x - 4)))/(2*(C1*x - 1))), Eq(f(x), -(x + sqrt(x*(4*C1*x + x - 4)))/(2*(C1*x - 1)))]
```
I tried to put the solution same as returned by dsolve but it automatically expands and differ in testing.

Reply:
It's the dreaded 2-arg mul!

Reply:
> It's the dreaded 2-arg mul!

Sorry couldn't get what does it mean?

Reply:
@smichr knows what it means. The solution is to put extra brackets in. One of these:
```
(-(x + sqrt(x*(4*C1*x + x - 4))))/(2*(C1*x - 1)))
-((x + sqrt(x*(4*C1*x + x - 4)))/(2*(C1*x - 1))))
```
Unfortunately multiplication is not associative in sympy.

Reply:
> @smichr knows what it means. The solution is to put extra brackets in. One of these:
> 
> ```
> (-(x + sqrt(x*(4*C1*x + x - 4))))/(2*(C1*x - 1)))
> -((x + sqrt(x*(4*C1*x + x - 4)))/(2*(C1*x - 1))))
> ```
> 
> Unfortunately multiplication is not associative in sympy.

i tried but still it fails

Reply:
The solution needs to be written like this to stop the rationals from distributing:
```
sol = [Eq(f(x), (-x + sqrt(x*(4*C1*x + x - 4)))/(C1*x - 1)/2), Eq(f(x), -((x + sqrt(x*(4*C1*x + x - 4)))/(C1*x - 1))/2)] 
```